Stack/Opc.Ua.Core/Types/BuiltIn/Variant.cs

```diff
@@ -2647,14 +2647,17 @@ public Matrix(Array elements, BuiltInType builtInType, params int[] dimensions)
 
             if (dimensions != null && dimensions.Length > 0)
             {
-                ulong length = 1;
+                int length = 1;
 
                 for (int ii = 0; ii < dimensions.Length; ii++)
                 {
-                    length *= (ulong)dimensions[ii];
+                    checked
```

Comment:
I liked the solution before more, because we could calculate the maximum length and throw an own exception. The checked block will throw an generic System.OverflowException, which looks like an issue of the stack, rather than the caller

Reply:
I agree that it would have been better from the exception point of view, but the reality is that the existing implementation is fragile against an overflow triggered by hand-crafted dimensions.
A solution could be to catch the System.OverflowExceptionand and rethrow our own ArgumentException exception at that time.

Reply:
This would be better :)